tiled/server/authentication.py

```diff
@@ -823,6 +824,40 @@ async def principal_list(
     return json_or_msgpack(request, principals)
 
 
+@base_authentication_router.post(
+    "/principal",
+    response_model=schemas.Principal,
+)
+async def create_service_principal(
+    request: Request,
+    principal=Security(get_current_principal, scopes=["read:principals"]),
+    db=Depends(get_database_session),
+    role: str = Query(...),
```

Comment:
My thinking was:
* The role name will never be large (i.e. it will fit in the URL always).
* The role name is not sensitive (i.e. not a credential).
* Query params are simple.

I guess JSON bodies are also simple, though. Is there a particular driving use case?

Reply:
I was thinking only for flexibility, but I agree that there is not a strong case for changing at the moment.